aws/rust-runtime/aws-config/src/environment/credentials.rs

```diff
@@ -29,7 +29,12 @@ impl EnvironmentVariableCredentialsProvider {
             .get("AWS_SECRET_ACCESS_KEY")
             .or_else(|_| self.env.get("SECRET_ACCESS_KEY"))
             .map_err(to_cred_error)?;
-        let session_token = self.env.get("AWS_SESSION_TOKEN").ok();
+        let session_token = self
+            .env
+            .get("AWS_SESSION_TOKEN")
+            .ok()
+            .map(|token| token.trim().to_string())
+            .filter(|token| !token.is_empty());
```

Comment:
Is it worth it to refactor this so we allocate after the filter? We could even use `filter_map`
```
.filter_map(|token| {
  let t = token.trim();
  
  if token.is_empty() {
    None
  } else {
    Some(token.to_string())
  }
})
```

Reply:
It doesn't look like [Option](https://doc.rust-lang.org/stable/std/option/enum.Option.html) has `filter_map`. Could potentially map it to another option and then `flatten` though.

Reply:
I always forget I can't user iterator methods on options and results